libnopegl/src/glsl/helper_noise.glsl

```diff
@@ -85,6 +85,20 @@ float random_grad(uvec2 x, vec2 f)
  */
 float random_grad(uvec3 x, vec3 f)
 {
+    /*
+    Adapted from https://web.archive.org/web/20180616011332/http://www.scratchapixel.com/lessons/procedural-generation-virtual-worlds%20/perlin-noise-part-2
+
+    float value = u32tof32(hash(x));
+    float theta = acos(2.0*value - 1.0);
+    float phi = 2.0 * value * 3.14;
+    vec3 grad = vec3(
+        cos(phi) * sin(theta),
+        sin(phi) * sin(theta),
+        cos(theta)
+    );
```

Comment:
Technically all the bits of the hash are supposedly uniformly distributed so how about something like:
```glsl
uint h = hash(x);
float v0 = u32tof32(0x01010101 * (h       & 0xff));
float v1 = u32tof32(0x01010101 * (h >>  8 & 0xff));
float v2 = u32tof32(0x01010101 * (h >> 16 & 0xff));
return dot(vec3(v0, v1, v2), f);
```

This is untested but bear with me: if Perlin can get away with less than 4-bits precision for the vectors, we can most certainly do with 8. So my reasoning here is that out of the 32-bit hash, we pick 3x8-bit for each dimension of the vector (shifts and masks), then we scale it up to 32-bits (with `0x01010101`) so that it can be converted to a uniform [0-1] float by `u32tof32()`.

(I'm assuming this could also work in 2D and is probably less expensive).